demo/src/features/debug3d/write.rs

```diff
@@ -1,25 +1,101 @@
-use crate::features::debug3d::{Debug3dDrawCall, Debug3dRenderFeature};
-use rafx::api::{RafxResult, RafxVertexBufferBinding};
+rafx::declare_render_feature_write_job!();
+
+use rafx::api::RafxPrimitiveTopology;
+use rafx::api::RafxVertexBufferBinding;
+use rafx::framework::{VertexDataLayout, VertexDataSetLayout};
+
+/// Vertex format for vertices sent to the GPU
+#[derive(Clone, Debug, Copy, Default)]
+#[repr(C)]
+pub struct Debug3DVertex {
+    pub pos: [f32; 3],
+    pub color: [f32; 4],
+}
+
+lazy_static::lazy_static! {
+    pub static ref DEBUG_VERTEX_LAYOUT : VertexDataSetLayout = {
+        use rafx::api::RafxFormat;
+
+        VertexDataLayout::build_vertex_layout(&Debug3DVertex::default(), |builder, vertex| {
+            builder.add_member(&vertex.pos, "POSITION", RafxFormat::R32G32B32_SFLOAT);
+            builder.add_member(&vertex.color, "COLOR", RafxFormat::R32G32B32A32_SFLOAT);
+        }).into_set(RafxPrimitiveTopology::LineStrip)
+    };
+}
+
 use rafx::framework::{BufferResource, DescriptorSetArc, MaterialPassResource, ResourceArc};
-use rafx::nodes::{
-    FeatureCommandWriter, RenderFeature, RenderFeatureIndex, RenderJobWriteContext,
-    RenderPhaseIndex, RenderView, SubmitNodeId,
-};
-
-pub struct Debug3dCommandWriter {
-    pub(super) vertex_buffer: Option<ResourceArc<BufferResource>>,
-    pub(super) draw_calls: Vec<Debug3dDrawCall>,
-    pub(super) debug3d_material_pass: ResourceArc<MaterialPassResource>,
-    pub(super) per_view_descriptor_sets: Vec<Option<DescriptorSetArc>>,
+use rafx::nodes::{push_view_indexed_value, RenderViewIndex};
+
+#[derive(Debug)]
+pub struct Debug3DDrawCall {
+    first_element: u32,
+    count: u32,
+}
+
+pub struct FeatureCommandWriterImpl {
+    vertex_buffer: Option<ResourceArc<BufferResource>>,
+    draw_calls: Vec<Debug3DDrawCall>,
+    debug3d_material_pass: ResourceArc<MaterialPassResource>,
+    per_view_descriptor_sets: Vec<Option<DescriptorSetArc>>,
 }
 
-impl FeatureCommandWriter for Debug3dCommandWriter {
```

Comment:
Maybe Debug3dWriteJob would be more consistent with the extract and prepare job naming. (In fact you named one of the macros "declare_render_feature_write_job!()"